agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -242,6 +242,12 @@ std::string slave_thread::print_cmdu_types(const message::sUdsHeader *cmdu_heade
 
 bool slave_thread::work()
 {
+    if (!m_logger_configured) {
```

Comment:
It is not for recovery, it is for assigning the thread id to the log file.
It can't be done elsewhere. It needs the first line of code that the thread executes.